h/static/scripts/widget-controller.js

```diff
@@ -405,45 +345,14 @@ module.exports = function WidgetController(
     return thread().totalChildren;
   };
 
-  /**
-   * Return the vertical scroll offset for the document in order to position the
-   * annotation thread with a given `id` or $$tag at the top-left corner
-   * of the view.
-   */
-  function scrollOffset(id) {
-    var maxYOffset = document.body.clientHeight - window.innerHeight;
-    return Math.min(maxYOffset, visibleThreads.yOffsetOf(id));
-  }
-
-  /** Scroll the annotation with a given ID or $$tag into view. */
-  function scrollIntoView(id) {
-    var estimatedYOffset = scrollOffset(id);
-    window.scroll(0, estimatedYOffset);
-
-    // As a result of scrolling the sidebar, the target scroll offset for
-    // annotation `id` might have changed as a result of:
-    //
-    // 1. Heights of some cards above `id` changing from an initial estimate to
-    //    an actual measured height after the card is rendered.
-    // 2. The height of the document changing as a result of any cards heights'
-    //    changing. This may affect the scroll offset if the original target
-    //    was near to the bottom of the list.
-    //
-    // So we wait briefly after the view is scrolled then check whether the
-    // estimated Y offset changed and if so, trigger scrolling again.
-    scopeTimeout($scope, function () {
-      var newYOffset = scrollOffset(id);
-      if (newYOffset !== estimatedYOffset) {
-        scrollIntoView(id);
-      }
-    }, 200);
-  }
+  // Object on which <thread-list> exposes methods to manipulate the list
+  $scope.threadListCtrl = {};
```

Comment:
This seems like a nasty hack, and isn't a pattern I've seen us use anywhere else in the codebase.

Given that it is only used here to scroll the annotation into view in response to a  `BEFORE_ANNOTATION_CREATED` event, why not just move that event handler onto the child controller?